fe/fe-core/src/main/java/org/apache/doris/datasource/property/storage/HDFSProperties.java

```diff
@@ -0,0 +1,195 @@
+            required = false,
+            description = "The principal of the kerberos authentication.")
+    private String hdfsKerberosPrincipal = "";
+
+    @ConnectorProperty(names = {"hdfs.authentication.kerberos.keytab", "hadoop.kerberos.keytab"},
+            required = false,
+            description = "The keytab of the kerberos authentication.")
+    private String hdfsKerberosKeytab = "";
+
+    @ConnectorProperty(names = {"hadoop.username"},
+            required = false,
+            description = "The username of Hadoop. Doris will user this user to access HDFS")
+    private String hadoopUsername = "";
+
+    @ConnectorProperty(names = {"hadoop.config.resources"},
+            required = false,
+            description = "The xml files of Hadoop configuration.")
+    private String hadoopConfigResources = "";
+
+    @ConnectorProperty(names = {"hdfs.impersonation.enabled"},
+            required = false,
+            supported = false,
+            description = "Whether to enable the impersonation of HDFS.")
+    private boolean hdfsImpersonationEnabled = false;
+
+    @ConnectorProperty(names = {"fs.defaultFS"}, required = false, description = "")
+    private String fsDefaultFS = "";
+
+    /**
+     * The final HDFS configuration map that determines the effective settings.
+     * Priority rules:
+     * 1. If a key exists in `overrideConfig` (user-provided settings), its value takes precedence.
+     * 2. If a key is not present in `overrideConfig`, the value from `hdfs-site.xml` or `core-site.xml` is used.
+     * 3. This map should be used to read the resolved HDFS configuration, ensuring the correct precedence is applied.
+     */
+    Map<String, String> finalHdfsConfig;
+
+    public HDFSProperties(Map<String, String> origProps) {
+        super(Type.HDFS, origProps);
+        loadFinalHdfsConfig(origProps);
+    }
+
+    public static boolean guessIsMe(Map<String, String> props) {
+        if (MapUtils.isEmpty(props)) {
+            return false;
+        }
+        if (props.containsKey("hadoop.config.resources") || props.containsKey("hadoop.security.authentication")) {
+            return true;
+        }
+        return false;
+    }
+
+    private void loadFinalHdfsConfig(Map<String, String> origProps) {
+        if (MapUtils.isEmpty(origProps)) {
+            return;
+        }
+        finalHdfsConfig = new HashMap<>();
+        origProps.forEach((key, value) -> {
+            if (key.startsWith("hadoop.") || key.startsWith("dfs.") || key.equals("fs.defaultFS")) {
+                finalHdfsConfig.put(key, value);
+            }
+        });
+
+    }
+
+    @Override
+    protected String getResourceConfigPropName() {
+        return "hadoop.config.resources";
+    }
+
+    protected void checkRequiredProperties() {
+        super.checkRequiredProperties();
+        checkConfigFileIsValid(hadoopConfigResources);
```

Comment:
before calling `checkRequiredProperties()`, we already call `loadConfigFromFile()`.
Which means for creation, we load config from file twice.